services/audit-service/src/main/java/datawave/microservice/audit/replay/ReplayController.java

```diff
@@ -0,0 +1,726 @@
+                if (status.getState() == CREATED) {
+                    RunningReplay replay = start(status);
+                    if (replay != null) {
+                        runningReplays.put(status.getId(), replay);
+                        replaysStarted++;
+                    }
+                }
+            }
+        }
+        
+        log.info(replaysStarted + " audit replays started");
+        return replaysStarted + " audit replays started";
+    }
+    
+    /**
+     * Gets the status of an audit replay
+     *
+     * @param id
+     *            The audit replay id
+     * @return the status of the audit replay
+     */
+    @ApiOperation(value = "Gets the status of an audit replay.")
+    @RolesAllowed({"AuthorizedUser", "AuthorizedServer", "InternalUser", "Administrator", "JBossAdministrator"})
+    @RequestMapping(path = "/{id}/status", method = org.springframework.web.bind.annotation.RequestMethod.GET)
+    public Status status(@ApiParam("The audit replay id") @PathVariable("id") String id, HttpServletResponse response) throws IOException {
+        
+        log.info("Getting status for audit replay with id " + id);
+        
+        Status status = status(id, replayProperties.isPublishEvents());
+        
+        if (status == null)
+            response.sendError(HttpServletResponse.SC_BAD_REQUEST, "No audit replay found with id " + id);
+        
+        return status;
+    }
+    
+    private Status status(String id, boolean publishEnabled) {
+        Status status = statusCache.retrieve(id);
+        if (status != null)
+            return idleCheck(status, publishEnabled);
+        return null;
+    }
+    
+    private Status idleCheck(Status status, boolean publishEnabled) {
+        // if the replay is RUNNING, and this hasn't been updated within the timeout interval, set the state to IDLE, and send out a stop request
+        if (status.getState() == RUNNING && (System.currentTimeMillis() - status.getLastUpdated().getTime()) > replayProperties.getIdleTimeoutMillis()) {
+            status.setState(IDLE);
+            statusCache.update(status);
+            stop(status, publishEnabled);
+        }
+        
+        return status;
+    }
+    
+    /**
+     * Lists the status for all audit replays
+     *
+     * @return list of statuses for all audit replays
+     */
+    @ApiOperation(value = "Lists the status for all audit replays.")
+    @RolesAllowed({"AuthorizedUser", "AuthorizedServer", "InternalUser", "Administrator", "JBossAdministrator"})
+    @RequestMapping(path = "/statusAll", method = org.springframework.web.bind.annotation.RequestMethod.GET)
+    public List<Status> statusAll() {
+        
+        log.info("Getting status for all audit replays");
+        
+        List<Status> statuses = statusCache.retrieveAll();
+        if (statuses != null)
+            statuses = statuses.stream().map(status -> idleCheck(status, replayProperties.isPublishEvents())).collect(Collectors.toList());
```

Comment:
It looks like idleCheck can never actually return a different status--it only has side effects. If that's the case, you could avoid duplicate list creation (and have more concise code) with:
```
if (statuses != null)
    statuses.forEach(status -> idleCheck(status, replayProperties.isPublishEvents());

Reply:
Good catch.  Updated in the next commit.